datafusion/common/src/stats.rs

```diff
@@ -258,6 +258,26 @@ impl Statistics {
         self
     }
 
+    /// Project the statistics to the given column indices.
+    ///
+    /// For example, if we had statistics for columns `{"a", "b", "c"}`,
+    /// projecting to `vec![2, 1]` would return statistics for columns `{"c",
+    /// "b"}`.
+    pub fn project(mut self, projection: Option<&Vec<usize>>) -> Self {
```

Comment:
Should this method also be used when we project the statistics in hash_join? 

https://github.com/alamb/datafusion/blob/86690fda27ddf7b983cf5597f5bde26ec70a725d/datafusion/physical-plan/src/joins/hash_join.rs#L790

Reply:
Makes sense, also this one seems a bit more efficient than the one in hash join.

Reply:
Done in 90aa0bd1c